### ISSUE TITLE
`reprun`: Improve handling of sorting when command uses `by`

### ISSUE THREAD

**Issue by bbdaniels** [worldbank/repkit#60]

**Comment by ankritisingh** [comment]
```
sysuse census, clear

gen check = 1

bys region: egen assigned 		= total(check)
bys region: egen treated_tot 	= total(check & region != 1)

recode region (1 = 2) (2 = 1) (3 = 4) (4 = 3)

* estpose is not flagged without recode (also nto sure why they used recode)
estpost tabstat assigned, statistics(mean) by(region) nototal

replace region = . if region == 1

* this is not flagged if the obs are not missing
bys region: sum pop


```
![Screenshot 2024-06-27 at 6 07 05 PM](https://github.com/worldbank/repkit/assets/54277703/ffc3dc8c-d508-4f91-9d09-19981649ad38)
![Screenshot 2024-06-27 at 6 07 14 PM](https://github.com/worldbank/repkit/assets/54277703/659cac26-f990-435b-b133-7307344f964b)

**Comment by bbdaniels** [comment]
I don't think this is solvable in a general case, especially when `by` is an option; I don't believe you can avoid the indeterminate within-group sort there. I have added a note on it in the manuscript. But I will continue to think about it...

**Comment by bbdaniels** [comment]
I had a good discussion about this at Stata Conference and they agreed it was likely to be unsolvable. In addition when `bys` is used in a user written command that leaves the data out of order it may or may not even be detectable by any kind of syntax checking. On the other hand, the flag is appropriate since unstable sorting will in fact lead to instability, and it is good to detect that even if there is no downstream consequence. So I advise to mark as `wontfix` and to write this out in documentation and in the manuscript.
